go/engine/upak_loader_test.go

```diff
@@ -330,3 +331,118 @@ func TestUPAKDeadlock(t *testing.T) {
 		t.Fatal("deadlocked!")
 	}
 }
+
+func TestLoadAfterAcctReset1(t *testing.T) {
+	// One context for user that will be doing LoadUser, and another
+	// for user that will sign up and reset itself.
+	tc := SetupEngineTest(t, "clu")
+	defer tc.Cleanup()
+
+	resetUserTC := SetupEngineTest(t, "clu2")
+	defer resetUserTC.Cleanup()
+
+	t.Logf("create new user")
+	fu := CreateAndSignupFakeUser(resetUserTC, "res")
+
+	fakeClock := clockwork.NewFakeClockAt(time.Now())
+	tc.G.SetClock(fakeClock)
+
+	loadUpak := func() error {
+		t.Logf("loadUpak: using username:%+v", fu.Username)
+		loadArg := libkb.NewLoadUserArg(tc.G)
+		loadArg.UID = fu.UID()
+		loadArg.PublicKeyOptional = false
```

Comment:
it doesn't pass with `PublicKeyOptional = true`, because in that case `CachedUPAKLoader.Load` returns `UserPlusAllKeys` with no keys instead of erroring out with `NoKeyError`.  I can add add another test that tries same thing but with `PublicKeyOptional = false` and sees if it still dismisses cache correctly when it sees nuked account.